android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -176,3 +153,104 @@ sealed class RemoteSettingsResult {
      */
     data class NetworkFailure(val error: Exception) : RemoteSettingsResult()
 }
+
+/**
+ * Data class representing serializable version of RemoteSettingsResponse.
+ */
+@Serializable
+private data class SerializableRemoteSettingsResponse(
+    val records: List<SerializableRemoteSettingsRecord>,
+    val lastModified: ULong,
+)
+
+/**
+ * Data class representing serializable version of RemoteSettingsRecord.
+ */
+@Serializable
+private data class SerializableRemoteSettingsRecord(
+    val id: String,
+    val lastModified: ULong,
+    val deleted: Boolean,
+    val attachment: SerializableAttachment?,
+    @Serializable(with = JSONObjectSerializer::class)
+    val fields: JSONObject,
```

Comment:
Shucks, I'm just finding that kotlinx does actually have a `JsonObject` [type](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-object/). 

I think it would be fine to keep the patch as is, but replacing the type here might let us get rid of the custom `JSONObjectSerializer`.

Reply:
I am guessing we still need to convert JsonObject to JSONObject , because both of the are a part of different library? I can look into this as a follow up